splitio/sync/split.py

```diff
@@ -318,3 +354,169 @@ def synchronize_splits(self, till=None):  # pylint:disable=unused-argument
 
         for split in to_delete:
             self._split_storage.remove(split)
+
+        return []
+
+    def _synchronize_json(self):
+        """
+        Update splits in storage for json mode.
+
+        :return: segment names string array
+        :rtype: [str]
+        """
+        try:
+            fetched, till = self._read_splits_from_json_file(self._filename)
+            segment_list = set()
+            fecthed_sha = util._get_sha(json.dumps(fetched))
+            if fecthed_sha != self._current_json_sha:
+                self._current_json_sha = fecthed_sha
+                if self._split_storage.get_change_number() <= till or till == self._DEFAULT_SPLIT_TILL:
+                    for split in fetched:
+                        if split['status'] == splits.Status.ACTIVE.value:
+                            parsed = splits.from_raw(split)
+                            self._split_storage.put(parsed)
+                            _LOGGER.debug("split %s is updated", parsed.name)
+                            segment_list.update(set(parsed.get_segment_names()))
+                        else:
+                            self._split_storage.remove(split['name'])
+
+                    self._split_storage.set_change_number(till)
+            return segment_list
+        except Exception as exc:
+            raise ValueError("Error reading splits from json.") from exc
+
+    def _read_splits_from_json_file(self, filename):
+        """
+        Parse a splits file and return a populated storage.
+
+        :param filename: Path of the file containing split
+        :type filename: str.
+
+        :return: Tuple: sanitized split structure dict, since and till
+        :rtype: Tuple(Dict, int, int)
+        """
+        try:
+            with open(filename, 'r') as flo:
+                parsed = json.load(flo)
+            santitized = self._sanitize_split(parsed)
+            flo.close
+            return santitized['splits'], santitized['till']
```

Comment:
this doesn't match rtype. Are we missing a dict or is the docstring outdated?

Reply:
fixed rtype